### ISSUE TITLE
Unusable mouse cursor in Windows resources

### ISSUE THREAD

**Issue by unsubtleguy** [dmacvicar/terraform-provider-libvirt#419]
#  Version Reports:

### Distro  version of host:

``` Fedora/NA ```

### Terraform Version Report
``` Terraform v0.11.8``` 

### Libvirt version
``` 4.1.0``` 

### terraform-provider-libvirt plugin version (git-hash)
``` c46f0f88a5b74319af7d1f1b219f1a7412f215de``` 
___
# Description of Issue/Question

When using Windows guests with the libvirt provider, by default the mouse input is completely unusable. I've had to manually work around by adding an evtouch input device (which results in being able to interact with the machine and use a mouse).

The bug is known, and is a KVM/libvirt problem rather than the provider - however I'd like to be able to not do this manually, given the role of Terraform.

Tl;dr:
* Terraform libvirt provider does not allow customisation of input devices
* When using modern (>2008) Windows(TM) Operating systems, when using the generic ps/2 input device, input is borderline unusable
  * By this I mean with great effort, you can move the mouse pointer around and use the machine - but the users mouse pointer does not align with the VMs mouse pointer
* However, when an `EvTouch USB Graphics Tablet`  input device is added, the mouse is responsive rather than not synchronising with the users mouse pointer

I've made a quick modification locally which appeared to work - and with quick testing, it added the evtouch device to deployed resources.

This would be of great benefit to anyone using this provider for Windows guests - and does not appear to harm Linux guests.

### Proposal

Ideally, the provider would just allow you to modify this on a per-resource basis, but the following appeared to work as a proof of concept:
```
$ git diff .
diff --git a/libvirt/domain_def.go b/libvirt/domain_def.go
index 09a41717..7a3aaa2f 100644
--- a/libvirt/domain_def.go
+++ b/libvirt/domain_def.go
@@ -8,6 +8,8 @@ import (
        libvirtxml "github.com/libvirt/libvirt-go-xml"
 )

+var tabletBus uint = 0
+
 // note source and target are not initialized
 func newFilesystemDef() libvirtxml.DomainFilesystem {
        return libvirtxml.DomainFilesystem{
@@ -35,6 +37,22 @@ func newDomainDef() libvirtxml.Domain {
                },
                CPU: &libvirtxml.DomainCPU{},
                Devices: &libvirtxml.DomainDeviceList{
+                               Inputs: []libvirtxml.DomainInput{
+                                       libvirtxml.DomainInput{
+                                               Type: "tablet",
+                                               Bus:  "usb",
+                                               Address: &libvirtxml.DomainAddress{
+                                                       USB: &libvirtxml.DomainAddressUSB{
+                                                               Bus:  &tabletBus,
+                                                               Port: "0.1",
+                                                       },
+                                               },
+                                       },
+                                       libvirtxml.DomainInput{
+                                               Type: "keyboard",
+                                               Bus:  "ps2",
+                                       },
+                               },
                        Graphics: []libvirtxml.DomainGraphic{
                                {
                                        Spice: &libvirtxml.DomainGraphicSpice{
```
___

**Comment by MalloZup** [comment]
@unsubtleguy  ciao thx for the issue. 

I think the most non-intruisive solution is give the user a possibility to specify  the DomainInput as variable.

**Comment by dmacvicar** [comment]
I think I would prefer to solve this like the xml/xslt filter that I suggested in #350 ([details](https://github.com/dmacvicar/terraform-provider-libvirt/issues/350#issuecomment-416634118) )

**Comment by MalloZup** [comment]
@dmacvicar yes, i think the future for all this extra-request parameters is to add `datasources` and have a generic solution that handle all possible data

**Comment by dmacvicar** [comment]
> @dmacvicar yes, i think the future for all this extra-request parameters is to add `datasources` or similar mechanism and have a generic solution that handle all possible data , so we don"t maintain all this fields

I was thinking on allowing referencing a xslt file to transform the final _xml_ generated by the provider, just before is loaded into libvirt, plus adding some examples.

**Comment by MalloZup** [comment]
@unsubtleguy fixed by this https://github.com/dmacvicar/terraform-provider-libvirt/tree/master/examples/xslt you can add custom parameters.  Look also at the doc. Thx
